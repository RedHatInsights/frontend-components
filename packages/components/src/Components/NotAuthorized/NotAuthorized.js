import React from 'react';
import PropTypes from 'prop-types';

import {
    Title,
    Button,
    EmptyState,
    EmptyStateVariant,
    EmptyStateIcon,
    EmptyStateBody
} from '@patternfly/react-core';

import { LockIcon } from '@patternfly/react-icons';

import './NotAuthorized.scss';

const NotAuthorized = ({ serviceName, icon: Icon, description, showReturnButton, className, ...props }) => {
    return (
        <EmptyState
            variant={ EmptyStateVariant.full }
            className={`ins-c-not-authorized ${className || ''}`}
            {...props}
        >
            <EmptyStateIcon icon={ Icon } />
            <Title
                headingLevel="h5"
                size="lg"
            >
                You do not have access to { serviceName }
            </Title>
            <EmptyStateBody>
                { description }
            </EmptyStateBody>
            {
                showReturnButton && (
                    document.referrer ?
                        <Button variant="primary" onClick={ () => history.back() }>Return to previous page</Button> :
                        <Button variant="primary" component="a" href=".">Go to landing page</Button>
                )
            }
        </EmptyState>
    );
};

NotAuthorized.propTypes = {
    serviceName: PropTypes.node,
    icon: PropTypes.func,
    description: PropTypes.node,
    showReturnButton: PropTypes.bool,
    className: PropTypes.string
};

NotAuthorized.defaultProps = {
    icon: LockIcon,
    description: 'Contact your organization administrator(s) for more information.',
    showReturnButton: true
};

export default NotAuthorized;
