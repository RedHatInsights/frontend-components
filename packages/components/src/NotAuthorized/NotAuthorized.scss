.ins-c-not-authorized {
    .pf-c-title {
        max-width: 540px;
        margin-left: auto;
        margin-right: auto;
    }
}
