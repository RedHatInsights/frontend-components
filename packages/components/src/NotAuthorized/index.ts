export * from '@patternfly/react-component-groups/dist/dynamic/UnauthorizedAccess';
export { default } from './NotAuthorized';
export { default as NotAuthorized } from './NotAuthorized';
