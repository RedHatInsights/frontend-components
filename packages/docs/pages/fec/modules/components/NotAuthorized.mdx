
# NotAuthorized

## Props

|name|type|default|description|
|----|----|-------|-----------|
|serviceName|`string`|||
|icon|React.ComponentType&lt;any&gt;|`LockIcon`||
|description|React.ReactNode|`<ContactBody />`||
|showReturnButton|<code>&#123;<br/><span className="default-prop-spacer"></span><span className="default-prop-spacer"></span><span className="default-prop-spacer"></span> "name": "boolean"<br />&#125;</code>|`true`||
|className|`string`|||
|title|React.ReactNode|||
|actions|React.ReactNode|`null`||
|prevPageButtonText|React.ReactNode|`'Return to previous page'`||
|toLandingPageText|React.ReactNode|`'Go to landing page'`||



