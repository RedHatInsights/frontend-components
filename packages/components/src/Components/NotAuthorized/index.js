export { default as NotAuthorized } from './NotAuthorized';
import './NotAuthorized.scss';
