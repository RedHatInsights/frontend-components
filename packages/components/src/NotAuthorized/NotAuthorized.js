import React from 'react';
import PropTypes from 'prop-types';

import { Title, Button, EmptyState, EmptyStateVariant, EmptyStateIcon, EmptyStateBody } from '@patternfly/react-core';

import { LockIcon } from '@patternfly/react-icons';

import './NotAuthorized.scss';

const ContactBody = () => (
  <React.Fragment>
    Contact your organization administrator(s) for more information or visit&nbsp;
    <a href={`./settings/my-user-access`}>My User Access</a>&nbsp; to learn more about your permissions.
  </React.Fragment>
);

const NotAuthorized = ({
  prevPageButtonText,
  toLandingPageText,
  title,
  actions,
  serviceName,
  icon: Icon,
  description,
  showReturnButton,
  className,
  ...props
}) => {
  const heading = title || `You do not have access to ${serviceName}`;
  return (
    <EmptyState variant={EmptyStateVariant.full} className={`ins-c-not-authorized ${className || ''}`} {...props}>
      <EmptyStateIcon icon={Icon} />
      <Title headingLevel="h5" size="lg">
        {heading}
      </Title>
      <EmptyStateBody>{description}</EmptyStateBody>
      {actions}
      {showReturnButton &&
        (document.referrer ? (
          <Button variant="primary" onClick={() => history.back()}>
            {prevPageButtonText}
          </Button>
        ) : (
          <Button variant="primary" component="a" href=".">
            {toLandingPageText}
          </Button>
        ))}
    </EmptyState>
  );
};

const serviceNamePropType = (props, propName, componentName, ...args) => {
  if (typeof props.title === 'undefined') {
    return PropTypes.node.isRequired(props, propName, componentName, ...args);
  }
};

NotAuthorized.propTypes = {
  serviceName: serviceNamePropType,
  icon: PropTypes.func,
  description: PropTypes.node,
  showReturnButton: PropTypes.bool,
  className: PropTypes.string,
  title: PropTypes.node,
  actions: PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf(PropTypes.node)]),
  prevPageButtonText: PropTypes.node,
  toLandingPageText: PropTypes.node,
};

NotAuthorized.defaultProps = {
  icon: LockIcon,
  showReturnButton: true,
  description: <ContactBody />,
  actions: null,
  prevPageButtonText: 'Return to previous page',
  toLandingPageText: 'Go to landing page',
};

export default NotAuthorized;
