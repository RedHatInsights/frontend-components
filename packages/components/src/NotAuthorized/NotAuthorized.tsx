import React from 'react';

import { Button, EmptyState, EmptyStateIcon, EmptyStateBody, EmptyStateProps, EmptyStateVariant, Title } from '@patternfly/react-core';

import { LockIcon } from '@patternfly/react-icons';

import './NotAuthorized.scss';

export interface NotAuthorizedProps extends Omit<EmptyStateProps, 'children' | 'title'> {
  serviceName?: string;
  icon?: React.ComponentType<any>;
  description?: React.ReactNode;
  showReturnButton?: boolean;
  className?: string;
  title?: React.ReactNode;
  actions?: React.ReactNode;
  prevPageButtonText?: React.ReactNode;
  toLandingPageText?: React.ReactNode;
}

const ContactBody = () => (
  <React.Fragment>
    Contact your organization administrator(s) for more information or visit&nbsp;
    <a href={`./settings/my-user-access`}>My User Access</a>&nbsp; to learn more about your permissions.
  </React.Fragment>
);

const NotAuthorized: React.FunctionComponent<NotAuthorizedProps> = ({
  prevPageButtonText = 'Return to previous page',
  toLandingPageText = 'Go to landing page',
  title,
  actions = null,
  serviceName,
  icon: Icon = LockIcon,
  description = <ContactBody />,
  showReturnButton = true,
  className,
  ...props
}) => {
  const heading = title || `You do not have access to ${serviceName}`;
  return (
    <EmptyState variant={EmptyStateVariant.full} className={`ins-c-not-authorized ${className || ''}`} {...props}>
      <EmptyStateIcon icon={Icon} />
      <Title headingLevel="h5" size="lg">
        {heading}
      </Title>
      <EmptyStateBody>{description}</EmptyStateBody>
      {actions}
      {showReturnButton &&
        (document.referrer ? (
          <Button variant="primary" onClick={() => history.back()}>
            {prevPageButtonText}
          </Button>
        ) : (
          <Button variant="primary" component="a" href=".">
            {toLandingPageText}
          </Button>
        ))}
    </EmptyState>
  );
};

export default NotAuthorized;
