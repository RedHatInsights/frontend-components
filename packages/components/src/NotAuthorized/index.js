export { default } from './NotAuthorized';
export { default as NotAuthorized } from './NotAuthorized';
import './NotAuthorized.scss';
