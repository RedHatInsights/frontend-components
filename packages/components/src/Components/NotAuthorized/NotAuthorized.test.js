import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NotAuthorized from './NotAuthorized';

describe('NotAuthorized component', () => {
    it('should render', () => {
        const wrapper = shallow(<NotAuthorized serviceName='Foo' />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
