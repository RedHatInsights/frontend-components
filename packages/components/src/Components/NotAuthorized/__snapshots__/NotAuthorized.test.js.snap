// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotAuthorized component should render 1`] = `
<Fragment>
  <Component
    className="ins-c-not-authorized"
    variant="full"
  >
    <Component
      icon={[Function]}
    />
    <Component
      headingLevel="h5"
      size="lg"
    >
       You do not have access to 
      Foo
    </Component>
    <Component>
      Contact your organization administrator(s) for more information.
    </Component>
    <Component
      component="a"
      href="."
      variant="primary"
    >
      Go to landing page
    </Component>
  </Component>
</Fragment>
`;
